features/support/fake_stripe.rb

```diff
@@ -0,0 +1,10 @@
+# Adapter to sit between the app and stripe for testing purposes
+class FakeStripe
+  def create_charge(**kwargs)
+    charges << kwargs
+  end
+
+  def charges
+    @charges ||= []
+  end
```

Comment:
@bhaibel - This is the kind of seam I'm talking about re: making a tiny wrapper that acts as a persistent spy that we can stub/etc without requiring us to spy explicitly; while still giving us the flexibility to stub with particular behaviors when we want to. Basically a pass-through adapter that in our test code shovels data. Ideally, we could pull this kind of thing out into a `stripe-spec` gem similar to [https://github.com/email-spec/email-spec](https://github.com/email-spec/email-spec)

Reply:
Yeah, this is the kind of thing I really don't like, it feels like a sink for the kind of thing that's the illusion of productivity -- I like the pass-through, it makes it better, and given that this is a lesser-of-evils situation maybe it makes it better enough, but I'm still pretty frowny-face about this tbh.

Lemme poke at how much of a pain it is to do test data cleanup via the Stripe API -- I feel like if we freshly create users on a per-test-run-basis rather than have fixture users, that gets around my concerns re too many secrets needed to run tests. It'll be slow but, eh, these are already feature tests.

Reply:
Let's hit the real stripe API then. I've had the literal opposite experience that you've had though, where using the pattern of explicitly stubbing interactions with external APIs in feature tests took a test suite that was full of brittleness and errors which degraded trust to one that could be relied on to actually give useful feedback.

That said, having a "contract/smoke test" where we actually go through the UI/API without any stubs is a really useful/good thing to have; especially if it can be fired against staging.

Can we put the stuff that actually hits real APIs in a different build that the features? That way the feature tests can stay reasonably independent and then the smoke/contract tests can be ran pre-promotion from staging to production?